### PR TITLE
feat: add Node.js chatbot recipe

### LINKED ISSUE #2147: Add chatbot-nodejs to the catalog
### Is your enhancement related to a problem? Please describe

A new chatbot-nodejs has landed into ai-recipes

See https://github.com/containers/ai-lab-recipes/tree/main/recipes/natural_language_processing/chatbot-nodejs

### Describe the solution you'd like

Add the recipe into the catalog

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mhdawson**
### What does this PR do?

NOTE: this should land after - https://github.com/containers/ai-lab-recipes/pull/822

Adds the Node.js chatbot example to the recipes included in podman-ai-lab

### What issues does this PR fix or reference?

JavaScript and TypeScript are often the second language supported by leading Libraries like Langchain, llamaindex and ollama indicating AI community believes they will be important for integrating AI into applications.  We should have at least a few Node.js/Javascript recipes.

Fixes #2147

### How to test this PR?

1) Clone podman-desktop repo, build as instructed in the contributing instrctions
2) Cone the repo/branch for this pull request  in podman-desktop-extension-ia-lab a the same level as 1)
3) Build podman-desktop-extension-ai-lab as indicated in the contributing info
4) Start podman-desktop as indicated in the contributing info for podman-desktop-extension-ai-lab
5) Go to the recipes and select and start the Node.js chatbot recipe.
6) Once everything is started validate that you can use the link on the details page for the running instance to get to the UI for the chatbot.

### CODE REVIEW

**Comment by mhdawson** [comment]
In case it matters, as an FYI I did talk to @slemeur about adding this recipe in advance.

**Review by Gregory-Pereira — APPROVED** [review]
/lgtm from the upstream ai-lab perspective

**Comment by mhdawson** [comment]
Just realised this will need an update for ` "ref": "v1.3.0",` to be updated to one that includes the fix in the ai-lab-recipies repo. Either that or point to a specific sha as the java one does ?

File: packages/backend/src/assets/ai.json [hunk 1]
```diff
@@ -86,6 +86,23 @@
       ],
       "backend": "llama-cpp"
     },
+    {
+      "id": "chatbot-nodejs",
+      "description": "This is a Node.js based chat demo application.",
+      "name": "Node.js based ChatBot",
+      "repository": "https://github.com/containers/ai-lab-recipes",
+      "ref": "v1.3.3",
```
Comment (mhdawson) [inline packages/backend/src/assets/ai.json:94]:
```suggestion
      "ref": "v1.3.0",
```

File: packages/backend/src/assets/ai.json [hunk 2]
```diff
@@ -86,6 +86,23 @@
       ],
       "backend": "llama-cpp"
     },
+    {
+      "id": "chatbot-nodejs",
+      "description": "This is a Node.js based chat demo application.",
+      "name": "Node.js based ChatBot",
+      "repository": "https://github.com/containers/ai-lab-recipes",
+      "ref": "v1.3.0",
```
Comment (feloy) [inline packages/backend/src/assets/ai.json:94]: I don't see the sources at the tag 1.3.0. I suppose it should be `1.3.0.1`

**Review by feloy — CHANGES_REQUESTED** [review]

File: packages/backend/src/assets/ai.json (hunk 2 above)
Comment (jeffmaury) [reply packages/backend/src/assets/ai.json:94]: Sorry I rename the tag (as I wrongly named it) and the latest is now v1.3.3.1

Comment (jeffmaury) [reply packages/backend/src/assets/ai.json:94]: But because of the fix I think it need a new tag

**Comment by mhdawson** [comment]
This one last PR needs to land, and then I'll update this PR to point to the release which contains that update https://github.com/containers/ai-lab-recipes/pull/825

File: packages/backend/src/assets/ai.json (hunk 2 above)
Comment (mhdawson) [reply packages/backend/src/assets/ai.json:94]: as commented below, we need this PR to land https://github.com/containers/ai-lab-recipes/pull/825 and then a new tag and then we can update this PR to point to that tag.

**Comment by Gregory-Pereira** [comment]
Just merged 825, /unhold

**Comment by jeffmaury** [comment]
> This one last PR needs to land, and then I'll update this PR to point to the release which contains that update [containers/ai-lab-recipes#825](https://github.com/containers/ai-lab-recipes/pull/825)

You can use commit SHA instead of tag for the reference, we'll use the tag when we will release the extension

**Comment by mhdawson** [comment]
@jeffmaury, @feloy updated to use the SHA. Should be good to go now assuming you will squash the commits as it is landed.

File: packages/backend/src/assets/ai.json [hunk 3]
```diff
@@ -86,6 +86,23 @@
       ],
       "backend": "llama-cpp"
     },
+    {
+      "id": "chatbot-nodejs",
```
Comment (jeffmaury) [inline packages/backend/src/assets/ai.json:90]:
```suggestion
      "id": "chatbot-nodejs-react",
```

as we are now usin the pattern feat-language-framemork now

**Review by jeffmaury — APPROVED** [review]
LGTM

**Review by slemeur — APPROVED** [review]

**Comment by jeffmaury** [comment]
@mergify backport 1.3.x

**Pull request merged**